app/dojo.py

```diff
@@ -0,0 +1,279 @@
+        sub_heading_unallocated = "\nPERSON NAME\t\tROLE\n"
+
+        return(heading_unallocated_offices, heading_unallocated_livingspaces,
+         sub_heading_unallocated)
+
+    def print_allocations(self, filename=""):
+        headings = self.choose_headings("print_allocations")
+        heading_office = headings[0]
+        sub_heading_office = headings[1]
+        heading_livingspace = headings[2]
+        sub_heading_livingspace = headings[3]
+
+        cprint(heading_office,  "blue")
+        for room in self.rooms:
+            if room.category == "Office":
+                cprint(sub_heading_office % room.name.upper(), "yellow")
+                for occupant in room.occupants:
+                    print("%s \t\t%s\t\t%s" %(str(occupant), 
+                        occupant.category.upper(), str(occupant.accomodation)))
+
+        cprint(heading_livingspace, "blue")
+        for room in self.rooms:
+            if room.category == "Living Space":
+                cprint(sub_heading_livingspace % room.name.upper(),  "yellow")
+                for occupant in room.occupants:
+                    print("%s \t\t%s\t\t%s" %(str(occupant), 
+                        occupant.category.upper(), str(occupant.office)))
+                            
+        if filename:
+            with open(filename, "w") as outputfile:
+                outputfile.write(heading_office)
+                for room in self.rooms:
+                    if room.category == "Office":
+                        outputfile.write(sub_heading_office % room.name.upper())
+                        for occupant in room.occupants:
+                            outputfile.write("%s \t\t%s\t\t%s\n" %(str(occupant), 
+                        occupant.category.upper(), str(occupant.accomodation)))
+
+                outputfile.write(heading_livingspace)
+                for room in self.rooms:
+                    if room.category == "Living Space":
+                        outputfile.write(sub_heading_livingspace % room.name.upper())
+                        for occupant in room.occupants:
+                            outputfile.write("%s \t\t%s\t\t%s\n" % (str(occupant), 
+                        occupant.category.upper(), str(occupant.office)))
+
+
+    def print_unallocated(self, filename=""):
+        headings = self.choose_headings("print_unallocated")
+        heading_office = headings[0]
+        heading_livingspace = headings[1]
+        sub_heading = headings[2]
+
+        cprint(heading_office,  "blue")
+        cprint(sub_heading,  "yellow")
+        for person in self.persons:
+            if person.office == "Unallocated":
+                print("%s\t\t%s" % (person.name, person.category.upper()))
+
+        cprint(heading_livingspace,  "blue")
+        cprint(sub_heading,  "yellow")
+        for person in self.persons:
+            if person.accomodation == "Unallocated":
+                print("%s\t\t%s" % (person.name, person.category.upper()))
+
```

Comment:
Instead of looping through the list of fellows and staff unallocated office and printing, just append to a string and print the string. Then use the same string and append it to a text file. This will reduce the number of loops you are making.